components/formats-gpl/src/loci/formats/in/MetamorphReader.java

```diff
@@ -1012,7 +1013,7 @@ else if (firstSeriesChannels != null) {
 
           if ((int) wave[waveIndex] >= 1) {
             // link LightSource to Image
-            int laserIndex = i * getEffectiveSizeC() + c;
+            int laserIndex = ((IMetadata)getMetadataStore()).getLightSourceCount(0);
```

Comment:
Is it safe to assume that ```getMetadataStore()``` always returns an ```IMetadata```?  Importing into OMERO in particular might break that assumption.